archive/frames/serializers.py

```diff
@@ -55,64 +57,74 @@ class FrameSerializer(serializers.ModelSerializer):
     url = serializers.CharField(read_only=True, help_text='File download URL associated with most recent version')
     filename = serializers.CharField(read_only=True, help_text='Full filename')
     area = PolygonField(allow_null=True, help_text='GeoJSON area that this frame covers')
-    DAY_OBS = serializers.DateField(input_formats=['iso-8601', '%Y%m%d'], help_text='Observation day in %Y%m%d format')
-    related_frames = serializers.PrimaryKeyRelatedField(
-        many=True,
-        queryset=Frame.objects.exclude(DATE_OBS=None),
-        required=False,
+    observation_day = serializers.DateField(input_formats=['iso-8601', '%Y%m%d'], help_text='Observation day in %Y%m%d format')
+    headers = serializers.JSONField(required=True, write_only=True)
+    configuration_type = serializers.ChoiceField(choices=get_configuration_type_tuples())
+    related_frame_filenames = serializers.ListField(
```

Comment:
Why have we switched to using filenames for related frames? Is this to avoid using specific keywords like L1IDFLAT as a related frame keyword? 

Does the ingester/ocs_archive library automatically look for related frame keywords and populate the list of filenames now?

Reply:
I'm don't think there was any switch here, we've always used filenames for the related frame names in the science archive I think? The keyword to filename conversion is here: https://github.com/observatorycontrolsystem/ocs_archive/blob/main/ocs_archive/input/headerdata.py#L52.